renderer/canvas.py

```diff
@@ -679,13 +684,44 @@ def _extract_stations_to_furniture(self, abstract_tile_matrix):
         Returns:
             asbtract_tile_matrix (List[List[String]]): matrix with furniture tiles added
         """
+
+        stations = []
         abstract_tile_matrix = [[abstract_tile_matrix[i][j] for j in range(len(abstract_tile_matrix[i]))]for i in range(len(abstract_tile_matrix))]
         for i, row in enumerate(self.layout):
             for j, col in enumerate(row):
                 if col is not None:
                     asset_info = self._choose_station_asset(col)
                     if asset_info["type"] == "tile":
                         abstract_tile_matrix[i][j] = asset_info["name"]
+                    else:
+                        name, _ = trim_item_ID(col)
+                        stations.append((i,j, name))
+        
+        directions = [(1,0), (0,1), (-1,0), (0, -1)]
```

Comment:
can we make this a helper function with docstring. explaining what is going on here for clarity